shape.cc

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <string>
+#include<cmath>
+#include <cstdlib>
+
+#define PI acos(-1)
+
+
+class Shape {
+  public:
+  Shape(std::string name) : name_{name} {}
+  std::string name() { return name_;}
+
+    virtual double area() {
+    std::cin>>width_;
+    std::cin>>height_;
+    area_= width_*height_;
+    return area_;
+    }
+
+  protected:
+    std::string name_;
+    double width_;
+    double height_;
+    double radius_;
+    double area_;
+
+};
+
+class Circle : public Shape {
+  public:
+    Circle(std::string name) : Shape(name) {}
```

Comment:
You need to have a radius
```suggestion
    Circle(const std::string& name, double radius) : Shape(name) {}
```